changelogs/unreleased/1638-add-soft-delete-option.yml

```
@@ -0,0 +1,8 @@
+description: >
+    Add ``soft_delete`` option to the exporter, to allow compiles to silently prevent resource
+    set deletion if the model is exporting resources included in this set.
```

I don't think this sentence will make a lot of sense to anyone who is no part of the core team.
```suggestion
    Add `--soft-delete` option to the exporter. This option makes sure that resource sets, specified using the `--delete-resource-set` option, are only deleted if there are no resources exported for this set.
```